### DISCUSSION TITLE
Frosted Glass / Half-transparent materials in gltf

### DISCUSSION THREAD

**Discussion by Bersaelor** [google/model-viewer#3993] [General]
Hey there,

I recently discovered some amazing advancements in Three.js, with regards to half transparent / frosted materials using the newly available `transmission`/`thickness`/`iOR` & clearcoat values.
Example pic:
![frosted dragon](https://user-images.githubusercontent.com/4517582/206679456-6014a57a-52c0-4fe5-bd29-2d2759fa15d0.jpg)
(Created using [this three.js demo](https://codesandbox.io/embed/15-q2nl8))
So far, I was only able to achieve this effect in Renderer like Blender's Cycles, it's amazing how performant the Three.js renderer is in the browser and how realistic it looks.

Now, I also looked through the [gltf 2.0 specifications](https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html) for similar parameters, atm the best I can do is using the alpha channel of the baseColor and set the `alphaMode` to `BLEND`. Thats how we used to do it.

Are there any options that model-viewer would render that I currently am not aware of?
Or are there any advancements in the pipeline with regard to transmission/ frosted materials?

**Comment by elalish** [comment]
Oh, in fact those were added to three.js in order to support the new glTF [extensions](https://www.khronos.org/blog/using-the-new-gltf-extensions-volume-index-of-refraction-and-specular), which are ratified now. Check out this [post](https://www.donmccurdy.com/2021/11/11/writing-volumetric-refraction-gltf/) for some more info on using them.

**Reply by elalish** [reply]
In short, they are all already supported by model-viewer: https://modelviewer.dev/fidelity/#khronos-DragonAttenuation
